### PR TITLE
Upgrade CMake to version 3.11.4

### PR DESCRIPTION
**Pull request by nickaleks**
### Description of the Change

This PR updates minimum CMake version requirements to 3.11.2. This allows using new features such as FetchContent.

### Benefits

More features, fewer bugs in CMake.

### Possible Drawbacks 

Some users will have to manually install CMake when trying to build iroha.

### CODE REVIEW

**Comment by dram** [comment]
Just FYI, cmake version in Ubuntu 18.04 LTS is 3.10.2.

https://packages.ubuntu.com/bionic/cmake

**Comment by nickaleks** [comment]
Yep, but we will need the functionality introduced in 3.11 in the future. Downloading and installing CMake is very straightforward, and all our development containers do that.

**Comment by dram** [comment]
Ok, np.

File: docker/android/Dockerfile [hunk 1]
```diff
@@ -19,11 +18,21 @@ ENV DEPS_DIR="/iroha/dependencies"
...
+    unzip zip build-essential automake libtool ca-certificates ccache zlib1g-dev libcurl4-openssl-dev libc6-dbg;  \
     rm -rf /var/lib/apt/lists/*
 
 RUN set -e; mkdir -p $DEPS_DIR/include $DEPS_DIR/lib
 
+# install cmake 3.11.4
+RUN set -e; \
+    git clone https://gitlab.kitware.com/cmake/cmake.git /tmp/cmake; \
+    (cd /tmp/cmake ; git checkout 316bd45439ad8ced6b31bcb10303a788038387ef); \
+    (cd /tmp/cmake ; /tmp/cmake/bootstrap --system-curl --parallel=${PARALLELISM} --enable-ccache); \
```
Comment (bakhtin) [inline docker/android/Dockerfile]: There is no `PARALLELISM` parameter defined. See other Docker files for how to setup it properly.

**Comment by bakhtin** [comment]
Android Docker image will soon be removed as we're migrating to a separate libiroha repo. Yet do we need to bump CMake version in that repo? Will we be using any new CMake functionality there?

**Comment by nickaleks** [comment]
Since it is not removed yet, we need to make sure the project can be built there.

**Pull request merged**